### ISSUE TITLE
[dev 5565] White flash when closing tab using X in TST sidebar

### ISSUE THREAD

**Issue by SXZ1** [piroor/treestyletab#1491]
Win8.1, Nightly 25-10-17. STR:

Create new profile, install TST dev, open new window, go to https://forums.geforce.com/default/board/110/ in the active tab
Open new tab, go to https://forums.geforce.com/default/board/27/support/
Press Ctrl-1, hover the mouse over the first tab in the sidebar and click X to close it

ER: Tab is closed, no white flash
AR: Tab is closed, white flash
Bug is not reproduced when using Ctrl-W or clicking X in the Firefox tab bar. This bug can also be reproduced multiple times without Firefox restart: just press Ctrl-Shift-T and try to close the tab again.

It looks like this bug is caused by https://bugzilla.mozilla.org/show_bug.cgi?id=1237654
Long story short: **toolkit.cosmeticAnimations.enabled false** disables animation, that causes white flash when closing tabs in stock Firefox/Nightly. The reason why this bug is also reproduced when using TST is that closing tab using X in TST sidebar doesn't trigger tab closing animation in Firefox tab bar.

So the solution is to trigger tab closing animation in Firefox tab bar after pressing X in TST sidebar.

**Comment by SXZ1** [comment]
I can still reproduce this bug with dev 5565. However, that's most likely Firefox bug. Could you confirm that, @piroor?

**Comment by piroor** [comment]
I've confirmed the white flash. I completely don't know what is the trigger...

**Comment by piroor** [comment]
Tab Center Redux also has this problem. Both TST and TCR closes the current tab by an WebExtensions API `browser.tabs.remove()`, and i've confirmed that the white flash appears when I just call the API from debug console. I think this is a problem of Firefox itself.

**Comment by piroor** [comment]
OK, I've found the reason why this problem happens only on addons. The WebExtensions API `browser.tabs.remove()` removes the tab immediately without animation.

https://dxr.mozilla.org/mozilla-central/rev/2535bad09d720e71a982f3f70dd6925f66ab8ec7/browser/components/extensions/ext-tabs.js#442

On the other hand, Ctrl-W and other operations on Firefox removes the tab with animation.

https://dxr.mozilla.org/mozilla-central/rev/2535bad09d720e71a982f3f70dd6925f66ab8ec7/browser/base/content/browser.js#2383
https://dxr.mozilla.org/mozilla-central/rev/2535bad09d720e71a982f3f70dd6925f66ab8ec7/browser/base/content/tabbrowser.xml#3083

I've confirmed that `gBrowser.removeTab(gBrowser.selectedTab)` in the browser console (Ctrl-Shift-J) also triggers the white flash but `gBrowser.removeTab(gBrowser.selectedTab, { animate: true })` doesn't.

**Comment by SXZ1** [comment]
Thank you very much! Could you leave a comment somewhere on bugzilla in order to pay attention to this Firefox bug? It looks like this thread is appropriate: https://bugzilla.mozilla.org/show_bug.cgi?id=1237654

I guess I'll close this bug.
